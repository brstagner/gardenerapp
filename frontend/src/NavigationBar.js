import { NavLink, useNavigate } from "react-router-dom";
import 'bootstrap/dist/css/bootstrap.min.css';
import 'bootstrap/dist/js/bootstrap.bundle.min.js';
import 'popper.js/dist/popper.min.js';
import './css/nav.css';


const NavigationBar = ({ currUser, logout }) => {

    const nav = useNavigate();

    const handleLogout = () => {
        logout();
        nav('/');
    };

    const admin =
        <div className="dropdown d-flex justify-content-center">
            <button
                className="btn btn-link nav-link dropdown-toggle"
                type="button"
                data-bs-toggle="dropdown"
                aria-expanded="false">
                Admin
            </button>
            <ul className="dropdown-menu">
                <li>
                    <NavLink
                        className="nav-link text-black"
                        to="/users">
                        User List
                    </NavLink>
                </li>
                <li>
                    <NavLink
                        className="nav-link text-black"
                        to="/gardens/all">
                        Garden List
                    </NavLink>
                </li>
                <li>
                    <NavLink
                        className="nav-link text-black"
                        to="/plants/all">
                        Plant List
                    </NavLink>
                </li>
            </ul>
        </div>;

    return (
        <nav className="navbar navbar-expand-sm navbar-custom">
            <a className="navbar-brand mb-0 h1 fs-2" href="/">Gardener</a>
            {currUser ? (
                <button
                    className="navbar-toggler"
                    type="button"
                    data-bs-toggle="collapse"
                    data-bs-target="#navbarNav"
                    aria-controls="navbarNav"
                    aria-expanded="false"
                    aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>) : null}
            {currUser ? (
                <div className="collapse navbar-collapse" id="navbarNav">
                    <ul className="navbar-nav ml-auto">
                        <li className="nav-item">
                            <NavLink
                                className="btn btn-link nav-link"
                                to="/profile">
                                Profile ({currUser.username})
                            </NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink
                                className="btn btn-link nav-link"
                                to="/gardens">
                                Gardens
                            </NavLink>
                        </li>
                        <li className="nav-item">
                            <NavLink
                                className="btn btn-link nav-link"
                                to="/plants">
                                Plants
                            </NavLink>
                        </li>
                        {currUser.isAdmin === "true" ? admin : null}
                        <li className="nav-item d-flex justify-content-center">
                            <button
                                className="btn btn-link nav-link"
                                onClick={handleLogout}>
                                Log Out
                            </button>
                        </li>
                    </ul>
                </div>) :
                null
            }
        </nav>
    );
};

export default NavigationBar;