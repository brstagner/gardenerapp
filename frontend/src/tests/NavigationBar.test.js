import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import NavigationBar from '../NavigationBar';

const user = { username: "user", userId: "1", isAdmin: "false", token: "user-token" };
const admin = { username: "admin", userId: "2", isAdmin: "true", token: "admin-token" };

// smoke test
it("renders without crashing", function () {
    render(
        <BrowserRouter >
            <NavigationBar currUser={user} />
        </BrowserRouter >);
    // screen.debug()
});

// snapshot test as non-admin user
it("matches snapshot for non-admin user", function () {
    const { asFragment } = render(
        <BrowserRouter >
            <NavigationBar currUser={user} />
        </BrowserRouter >);
    expect(screen.queryAllByText("Admin")).toHaveLength(0);
    expect(asFragment()).toMatchSnapshot();
});

// snapshot test as admin
it("matches snapshot for admin", function () {
    const { asFragment } = render(
        <BrowserRouter>
            <NavigationBar currUser={admin} />
        </BrowserRouter>);
    expect(screen.getByText("Admin")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
});