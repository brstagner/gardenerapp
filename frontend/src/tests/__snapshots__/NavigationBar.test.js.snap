// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for admin 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-sm navbar-custom"
  >
    <a
      class="navbar-brand mb-0 h1 fs-2"
      href="/"
    >
      Gardener
    </a>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-bs-target="#navbarNav"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        class="navbar-nav ml-auto"
      >
        <li
          class="nav-item"
        >
          <a
            class="btn btn-link nav-link"
            href="/profile"
          >
            Profile (admin)
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="btn btn-link nav-link"
            href="/gardens"
          >
            Gardens
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="btn btn-link nav-link"
            href="/plants"
          >
            Plants
          </a>
        </li>
        <div
          class="dropdown d-flex justify-content-center"
        >
          <button
            aria-expanded="false"
            class="btn btn-link nav-link dropdown-toggle"
            data-bs-toggle="dropdown"
            type="button"
          >
            Admin
          </button>
          <ul
            class="dropdown-menu"
          >
            <li>
              <a
                class="nav-link text-black"
                href="/users"
              >
                User List
              </a>
            </li>
            <li>
              <a
                class="nav-link text-black"
                href="/gardens/all"
              >
                Garden List
              </a>
            </li>
            <li>
              <a
                class="nav-link text-black"
                href="/plants/all"
              >
                Plant List
              </a>
            </li>
          </ul>
        </div>
        <li
          class="nav-item d-flex justify-content-center"
        >
          <button
            class="btn btn-link nav-link"
          >
            Log Out
          </button>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot for non-admin user 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-sm navbar-custom"
  >
    <a
      class="navbar-brand mb-0 h1 fs-2"
      href="/"
    >
      Gardener
    </a>
    <button
      aria-controls="navbarNav"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-bs-target="#navbarNav"
      data-bs-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="collapse navbar-collapse"
      id="navbarNav"
    >
      <ul
        class="navbar-nav ml-auto"
      >
        <li
          class="nav-item"
        >
          <a
            class="btn btn-link nav-link"
            href="/profile"
          >
            Profile (user)
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="btn btn-link nav-link"
            href="/gardens"
          >
            Gardens
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            class="btn btn-link nav-link"
            href="/plants"
          >
            Plants
          </a>
        </li>
        <li
          class="nav-item d-flex justify-content-center"
        >
          <button
            class="btn btn-link nav-link"
          >
            Log Out
          </button>
        </li>
      </ul>
    </div>
  </nav>
</DocumentFragment>
`;
